netlogo-gui/src/main/app/NotificationBanner.scala

```diff
@@ -0,0 +1,213 @@
+    messageLabel.setForeground(InterfaceColors.ANNOUNCEMENTS_BANNER_TEXT)
+    closeButton.setForeground(InterfaceColors.ANNOUNCEMENTS_BANNER_TEXT)
+    scrollPane.getHorizontalScrollBar.setBackground(InterfaceColors.DIALOG_BACKGROUND)
+    scrollPane.getVerticalScrollBar.setBackground(InterfaceColors.DIALOG_BACKGROUND)
+
+  }
+
+  // Method to fetch JSON content from a URL
+  private def fetchJsonFromUrl(): String = {
+    try {
+      val source = Source.fromURL(jsonUrl)
+      val content = source.mkString
+      source.close()
+      content
+    } catch {
+      case e: Exception =>
+      ""
+    }
+  }
+
+  // Method to parse JSON content to a list of JsonObject instances
+  private def parseJsonToList(jsonContent: String): List[JsonObject] = {
+    if(jsonContent.isEmpty){
+      return Nil
+    }
+
+      val parser = new JSONParser()
+      val jsonArray = parser.parse(jsonContent).asInstanceOf[JSONArray]
+      jsonArray.toArray.flatMap { obj =>
+        val jsonObject = obj.asInstanceOf[JSONObject]
+        Option(jsonObject.get("event-id").asInstanceOf[Long].toInt).map { eventId =>
+          val title = Option(jsonObject.get("title")).map(_.toString).getOrElse("")
+          val fullText = Option(jsonObject.get("fullText")).map(_.toString).getOrElse("")
+          val date = Option(jsonObject.get("date")).map(_.toString).getOrElse("")
+          JsonObject(eventId, date, title, fullText)
+        }
+      }.toList.sortBy(_.eventId)(Ordering[Int].reverse)
+
+  }
+
+  // Method to show JSON content in a dialog
+  private def showJsonInDialog(): Unit = {
+
+    val prefs = Preferences.userNodeForPackage(getClass)
+    try {
+      val jsonContent = fetchJsonFromUrl()
+      jsonObjectList = parseJsonToList(jsonContent) // Populate the class variable
+      val formattedString = formatJsonObjectList(jsonObjectList)
+      if(!isShowNeeded()){
+        return
+      }
+      val lastSeenEventId = prefs.getInt(lastSeenEventIdKey, -1); // Returns -1 if "event-id" is not found
+      if(jsonObjectList.head.eventId <= lastSeenEventId){
+        return
+      }
+      val html = InfoFormatter.toInnerHtml(formattedString)
+
+      if (!jsonContent.trim.isEmpty) {
+          editorPane = new JEditorPane {
+            self =>
+
+            setDragEnabled(false)
+            setEditable(false)
+            setContentType("text/html")
+            setOpaque(true)
+            setBackground(InterfaceColors.CODE_BACKGROUND) // Set the background color)
+            setForeground(InterfaceColors.DEFAULT_COLOR) //Set the font color
+            setText(html)
+            setCaretPosition(0)
+          }
+
+          scrollPane = new JScrollPane(editorPane)
+          scrollPane.setPreferredSize(new Dimension(500, 400))
+          val panel = new JPanel(new BorderLayout())
+          panel.add(scrollPane, BorderLayout.CENTER)
+          val options: Array[AnyRef] = Array(I18N.gui.get("common.buttons.ok"))
```

Comment:
Aren't these all `String`s?

Reply:
Marking this as resolved because it won't be needed once he converts to my theme-synchronized OptionPane class.